core/src/test/scala/akka/persistence/postgres/journal/PostgresJournalSpec.scala

```diff
@@ -54,6 +55,119 @@ abstract class PostgresJournalSpec(config: String, schemaType: SchemaType)
+
+      // when
+      writeSingleMessage(1, perId, sender.ref, writerUuid)
+
+      // then
+      val metadataExists = db.run(metadataTable.filter(_.persistenceId === perId).exists.result).futureValue
+      metadataExists shouldBe true
+    }
+
+    "upsert only max_sequence_number and max_ordering if metadata already exists" in {
+      // given
+      val perId = "perId-meta-2"
+      val sender = TestProbe()
+      writeSingleMessage(1, perId, sender.ref, writerUuid)
+      val prevMaxSeqNr =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxSequenceNumber).result.head).futureValue
+      val prevMaxOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxOrdering).result.head).futureValue
+      val prevMinOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.minOrdering).result.head).futureValue
+
+      // when
+      writeSingleMessage(2, perId, sender.ref, writerUuid)
+
+      // then
+      val newMaxSeqNr =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxSequenceNumber).result.head).futureValue
+      val newMaxOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxOrdering).result.head).futureValue
+      val newMinOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.minOrdering).result.head).futureValue
+
+      newMaxSeqNr shouldBe prevMaxSeqNr + 1
+      newMaxOrdering shouldBe prevMaxOrdering + 1
+      newMinOrdering shouldBe prevMinOrdering
+    }
+
+    "set min_ordering to -1 when no metadata entry exists but the event being inserted is not the first one for the persistenceId (sequence_number > 1)" in {
+      // given
+      val perId = "perId-meta-3"
+      val sender = TestProbe()
+      writeSingleMessage(1, perId, sender.ref, writerUuid)
+      val prevMaxSeqNr =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxSequenceNumber).result.head).futureValue
+      val prevMaxOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxOrdering).result.head).futureValue
+
+      // when
+      // simulate case where metadata does not exist, but persistenceId already has events
+      db.run(metadataTable.filter(_.persistenceId === perId).delete).futureValue
+      // write new event of same persistenceId
+      writeSingleMessage(2, perId, sender.ref, writerUuid)
+
+      // then
+      val newMaxSeqNr =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxSequenceNumber).result.head).futureValue
+      val newMaxOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.maxOrdering).result.head).futureValue
+      val newMinOrdering =
+        db.run(metadataTable.filter(_.persistenceId === perId).map(_.minOrdering).result.head).futureValue
+
+      newMaxSeqNr shouldBe prevMaxSeqNr + 1
+      newMaxOrdering shouldBe prevMaxOrdering + 1
+      newMinOrdering shouldBe -1
```

Comment:
suggestion: use a constant somewhere to name this `-1` flag? (`UNSET_MIN_ORDERING`?)

Reply:
The value is mainly used on the migrations that will insert the trigger and the `.sql` files we have for tests setups.

I've created a local variable close to the tests to enhance readability.